src/java/org/apache/cassandra/schema/ColumnMetadata.java

```diff
@@ -627,11 +716,24 @@ public ColumnMetadata deserialize(DataInputPlus in, Types types, UserFunctions f
             boolean masked = in.readBoolean();
             if (masked)
                 mask = ColumnMask.serializer.deserialize(in, ksName, type, types, functions, version);
-            return new ColumnMetadata(ksName, tableName, new ColumnIdentifier(nameBB, name), type, position, kind, mask);
+            ColumnConstraints constraints;
+            if (version.isAtLeast(Version.V6) && in.readBoolean())
+                constraints = ColumnConstraints.serializer.deserialize(in, version.asInt());
+            else
+                constraints = ColumnConstraints.NO_OP;
+            return new ColumnMetadata(ksName, tableName, new ColumnIdentifier(nameBB, name), type, position, kind, mask, constraints);
         }
 
         public long serializedSize(ColumnMetadata t, Version version)
         {
+            long constraintsSize = 0;
+            if (version.isAtLeast(Version.V6))
```

Comment:
How does our logic resolve / handle being in a mixed version cluster where some members are older versions and some newer? Does TCM keep the version at the lowest common denominator in that case?

Reply:
I think @yifan-c already touches on that on this comment
https://github.com/apache/cassandra/pull/3562#discussion_r1913673066